ir/attrs.h

```diff
@@ -3,10 +3,15 @@
 // Copyright (c) 2018-present The Alive2 Authors.
 // Distributed under the MIT license that can be found in the LICENSE file.
 
+#include "smt/exprs.h"
 #include <ostream>
 
 namespace IR {
 
+class State;
```

Comment:
state can be dropped now

Reply:
On my machine, it raised compile error due to the State argument in encode

Reply:
Oops, if the question was that State can be dropped from encode's argument, it is still necessary because encoding pointer attributes requires Memory.